### PR TITLE
REPO-4009:DBP SSO: Possible to use 'alf_ticket' with AIS SSO against V1 REST API

### PR DESCRIPTION
**Pull request by andrei-rebegea**

### CODE REVIEW

File: src/main/java/org/alfresco/rest/api/impl/AuthenticationsImpl.java [hunk 1]
```diff
@@ -183,18 +194,81 @@ protected String getTicket(Parameters parameters)
...
+            if (isBearerHeader)
+            {
+                return getTicketFormRemoteUserMapperUserId(parameters);
+            }
         }
-        return authObj.getTicket();
+        throw new InvalidArgumentException("Authorization not supported.");
     }
+
+    private String getTicketFormRemoteUserMapperUserId(Parameters parameters)
```
Comment (derekhulley) [inline src/main/java/org/alfresco/rest/api/impl/AuthenticationsImpl.java]: Typo?

File: src/main/java/org/alfresco/rest/api/impl/AuthenticationsImpl.java [hunk 2]
```diff
@@ -183,18 +194,81 @@ protected String getTicket(Parameters parameters)
...
+        {
+            final String decodedAuthorisation = new String(Base64.decode(authorizationParts[1]));
+            Authorization authObj = new Authorization(decodedAuthorisation);
+            if (!authObj.isTicket())
+            {
+                throw new InvalidArgumentException("Ticket base authentication required.");
+            }
+            return authObj.getTicket();
+        }
+        else
```
Comment (killerboot) [inline src/main/java/org/alfresco/rest/api/impl/AuthenticationsImpl.java]: Will it be better to use the following structure?
```
else if(isBearerHeader)
{
...
}
else
{
    throw new InvalidArgumentException("Authorization not supported.");
}
 ```

Comment (andrei-rebegea) [reply src/main/java/org/alfresco/rest/api/impl/AuthenticationsImpl.java]: I fixed that. Thanks

File: src/main/java/org/alfresco/rest/api/impl/AuthenticationsImpl.java (hunk 1 above)
Comment (derekhulley) [reply src/main/java/org/alfresco/rest/api/impl/AuthenticationsImpl.java]: "Form" "From"

Comment (andrei-rebegea) [reply src/main/java/org/alfresco/rest/api/impl/AuthenticationsImpl.java]: I will fix it. I may need new glasses :(

File: src/test/java/org/alfresco/rest/api/tests/AuthenticationsTest.java [hunk 3]
```diff
@@ -185,13 +196,44 @@ public void testCreateValidateDeleteTicket() throws Exception
 
         // Get list of site by appending the invalidated ticket
         getAll(SiteEntityResource.class, paging, ticket, 401);
-
-
+    }
+        /**
```
Comment (killerboot) [inline src/test/java/org/alfresco/rest/api/tests/AuthenticationsTest.java]: I suspect there is indentation problem somewhere, see the full file:
https://github.com/Alfresco/alfresco-remote-api/blob/10a5bbb1f34c7e0484ad4dfe1db100388fbb8628/src/test/java/org/alfresco/rest/api/tests/AuthenticationsTest.java

**Pull request merged**